component/plugincontroller/component.go

```diff
@@ -0,0 +1,34 @@
+package plugincontroller
+
+import (
+	"github.com/docker/docker/component"
+	"github.com/docker/docker/plugin"
+	"golang.org/x/net/context"
+)
+
+const name = "plugincontroller"
+
+// Set sets the plugin controller component
+func Set(s *plugin.Manager) (cancel func(), err error) {
+	return component.Register(name, s)
+}
```

Comment:
Yes, I agree that the coordinator should not import the subsystem packages. They should register themselves with the coordinator.